models/topic.go

```diff
@@ -9,3 +11,45 @@ type Topic struct {
 	Keywords    []string `bson:"keywords,omitempty"     json:"keywords,omitempty"`
 	State       string   `bson:"state,omitempty"        json:"state,omitempty"`
 }
+
+// Validate checks that a topic struct complies with the state constraints, if provided. !!! may want to add more in future
+func (t *Topic) Validate() error {
+
+	if _, err := ParseState(t.State); err != nil {
+		return apierrors.ErrTopicInvalidState
+	}
+
+	// !!! add other checks, etc
+	return nil
+}
+
+// ValidateTransitionFrom checks that this topic state can be validly transitioned from the existing state
+func (t *Topic) ValidateTransitionFrom(existing *Topic) error {
+
+	// check that state transition is allowed, only if state is provided
+	if t.State != "" {
```

Comment:
Should there be an initial check for `t.State == ""`? I assume this would be an invalid state?

Reply:
the Validate() funtion in topic.go checks that State exists in the document.
(this is a work in progress as there is more to do in future parts of the spec, so its fine for now)